docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,51 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow any
+kind of customisations since it would make the system even more complex to test
+and maintain over the time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases running on the top of it.
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted and
+resistant to various type of events such as node reboot, upgrade, downgrade,
+backup, restore
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- Number of replicas for the Prometheus and Alert Manager cluster
```

Comment:
Grafana could also get more replicas, no?

Reply:
Yes I can add it but should I also talk about grafana configurations? (dashboards, datasources ...)?

Reply:
Yes you could, if you think it's something we should support.

Reply:
ok adding it